bayesian/__init__.py

```diff
@@ -94,6 +106,27 @@ def access_control_allow_origin(response):
 
 app = create_app()
 
+# Add prometheus wsgi middleware to route /metrics requests
+app.wsgi_app = DispatcherMiddleware(app.wsgi_app, {
+    '/metrics': make_wsgi_app(METRICS_REGISTRY)
+})
+
+
+def log_request_started(sender, **args):  # noqa
+    """Request Start Signal."""
+    setattr(g, 'time_start', time.time())
+    sender.logger.debug('Request context is set up')
+
+
+def log_request_finished(sender, **extra):
+    """Request Finish Signal."""
+    emit_response_metrics(extra.get("response").status_code, METRICS)
+    sender.logger.debug('Request is finished', request.path)
+
+
+request_started.connect(log_request_started, app)
+request_finished.connect(log_request_finished, app)
```

Comment:
It is not a good approach to track every path, which would result in high cardinality output and waste of storage. Better track only the intended endpoints,  e.g. component-analysis.

Reply:
Ack